proposals/function-pointers.md

```diff
@@ -212,6 +212,29 @@ Restrictions of this feature:
   deliberately not specified by the language. This includes whether they are static vs. instance or
   exactly what signature they are emitted with.
 
+
+### Operators on Function Pointer Types
+
+The section in unsafe code on operators is modified as such:
+
+> In an unsafe context, several constructs are available for operating on all _pointer\_type_s that are not _funcptr\_type_s:
+>
+> *  The `*` operator may be used to perform pointer indirection ([Pointer indirection](unsafe-code.md#pointer-indirection)).
+> *  The `->` operator may be used to access a member of a struct through a pointer ([Pointer member access](unsafe-code.md#pointer-member-access)).
+> *  The `[]` operator may be used to index a pointer ([Pointer element access](unsafe-code.md#pointer-element-access)).
+> *  The `&` operator may be used to obtain the address of a variable ([The address-of operator](unsafe-code.md#the-address-of-operator)).
+> *  The `++` and `--` operators may be used to increment and decrement pointers ([Pointer increment and decrement](unsafe-code.md#pointer-increment-and-decrement)).
+> *  The `+` and `-` operators may be used to perform pointer arithmetic ([Pointer arithmetic](unsafe-code.md#pointer-arithmetic)).
+> *  The `==`, `!=`, `<`, `>`, `<=`, and `=>` operators may be used to compare pointers ([Pointer comparison](unsafe-code.md#pointer-comparison)).
+> *  The `stackalloc` operator may be used to allocate memory from the call stack ([Fixed size buffers](unsafe-code.md#fixed-size-buffers)).
+> *  The `fixed` statement may be used to temporarily fix a variable so its address can be obtained ([The fixed statement](unsafe-code.md#the-fixed-statement)).
+> 
+> In an unsafe context, several constructs are available for operating on all _funcptr\_type_s:
+> *  The `&` operator may be used to obtain the address of static methods ([Allow address-of to target methods](function-pointers.md#allow-address-of-to-target-methods))
+> *  The `==`, `!=`, `<`, `>`, `<=`, and `=>` operators may be used to compare pointers ([Pointer comparison](unsafe-code.md#pointer-comparison)).
```

Comment:
`<`, `>`, `<=`, and `=>` do not make sense for function pointers. I know that C/C++ allows it, but it is not necessarily a good prior art to follow.

Reply:
I wish we could block them, but they're defined on `void*`, and function pointers are implicitly convertible.

Reply:
`==` and `!=` are questionable too. Function pointers in .NET are not stable. For example, `ldftn` on same function can return different pointer values in the two places in the program. It does actually happen in practice, e.g. a typical example is `ldftn M` before M was executed for the first time and after M was executed for the first time.

Reply:
Always emit a warning when any of these are used on function pointers?

Reply:
Hmm, that's interesting. Seems like a good candidate for a warning then. I'll add a note to the tracking issue.